frontend/src/components/widgets/CameraInput/CameraInput.tsx

```diff
@@ -0,0 +1,579 @@
+import {
+  FileUploaderState as FileUploaderStateProto,
+  UploadedFileInfo as UploadedFileInfoProto,
+  CameraInput as CameraInputProto,
+} from "src/autogen/proto"
+
+import { FormClearHelper } from "src/components/widgets/Form"
+import { FileUploadClient } from "src/lib/FileUploadClient"
+import { WidgetStateManager } from "src/lib/WidgetStateManager"
+import { logError } from "src/lib/log"
+import {
+  WidgetLabel,
+  StyledWidgetLabelHelp,
+} from "src/components/widgets/BaseWidget"
+import TooltipIcon from "src/components/shared/TooltipIcon"
+import { Placement } from "src/components/shared/Tooltip"
+
+import {
+  UploadFileInfo,
+  UploadedStatus,
+  UploadingStatus,
+} from "../FileUploader/UploadFileInfo"
+
+import CameraInputButton from "./CameraInputButton"
+import WebcamComponent from "./WebcamComponent"
+
+import { StyledBox, StyledCameraInput, StyledSpan } from "./styled-components"
+
+export interface Props {
+  element: CameraInputProto
+  widgetMgr: WidgetStateManager
+  uploadClient: FileUploadClient
+  disabled: boolean
+  width: number
+}
+
+type FileUploaderStatus =
+  | "ready" // FileUploader can upload or delete files
+  | "updating" // at least one file is being uploaded or deleted
+
+interface State {
+  /**
+   * Base64-encoded image data of the current frame from the camera.
+   */
+  imgSrc: string | null
+
+  shutter: boolean
+
+  minShutterEffectPassed: boolean
+  /**
+   * List of files dropped on the FileUploader by the user. This list includes
+   * rejected files that will not be updated.
+   */
+  files: UploadFileInfo[]
+
+  /**
+   * The most recent file ID we've received from the server. This gets sent
+   * back to the server during widget update so that it clean up
+   * orphaned files. File IDs start at 1 and only ever increase, so a
+   * file with a higher ID is guaranteed to be newer than one with a lower ID.
+   */
+  newestServerFileId: number
+  /** Represents whether the component is in clear photo mode
+   * Time interval between `Clear Photo` button clicked and access to Webcam recived again
+   */
+  inClearPhotoMode: boolean
+}
+
+class CameraInput extends React.PureComponent<Props, State> {
+  private localFileIdCounter = 1
+
+  private RESTORED_FROM_WIDGET_STRING = "RESTORED_FROM_WIDGET"
+
+  private readonly formClearHelper = new FormClearHelper()
+
+  public constructor(props: Props) {
+    super(props)
+    this.state = this.initialValue
+  }
+
+  private getProgress = (): number | null | undefined => {
+    if (
+      this.state.files.length > 0 &&
+      this.state.files[this.state.files.length - 1].status.type === "uploading"
+    ) {
+      const status = this.state.files[this.state.files.length - 1]
+        .status as UploadingStatus
+      return status.progress
+    }
+    return undefined
+  }
+
+  private setInClearPhotoMode = (inClearPhotoMode: boolean): void => {
+    this.setState({ inClearPhotoMode })
+  }
+
+  private handleCapture = (imageSrc: string | null): void => {
```

Comment:
nit: would be good to rename this `imgSrc` so that the object construction shorthand can be used below